store/sqlstore/channel_store.go

```diff
@@ -783,53 +783,52 @@ func (s SqlChannelStore) get(id string, master bool, allowFromCache bool) (*mode
 }
 
 // Delete records the given deleted timestamp to the channel in question.
-func (s SqlChannelStore) Delete(channelId string, time int64) store.StoreChannel {
+func (s SqlChannelStore) Delete(channelId string, time int64) (*model.Channel, *model.AppError) {
 	return s.SetDeleteAt(channelId, time, time)
 }
 
 // Restore reverts a previous deleted timestamp from the channel in question.
-func (s SqlChannelStore) Restore(channelId string, time int64) store.StoreChannel {
+func (s SqlChannelStore) Restore(channelId string, time int64) (*model.Channel, *model.AppError) {
 	return s.SetDeleteAt(channelId, 0, time)
 }
 
 // SetDeleteAt records the given deleted and updated timestamp to the channel in question.
-func (s SqlChannelStore) SetDeleteAt(channelId string, deleteAt, updateAt int64) store.StoreChannel {
-	return store.Do(func(result *store.StoreResult) {
-		defer s.InvalidateChannel(channelId)
+func (s SqlChannelStore) SetDeleteAt(channelId string, deleteAt, updateAt int64) (*model.Channel, *model.AppError) {
```

Comment:
Probably here makes more sense to return just `*model.AppError` because in the previous code `result.Data` never was set, so... we weren't returning the "Data" before.